Software 0.31.2 release notes (#1816)
astronomer/docs: software/release-notes.md

0.31 is the latest stable version of Astronomer Software, while 0.30 remains the latest long-term support (LTS) release. To upgrade to 0.31, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.31.2

> Release date: February 1, 2023

Author: ```suggestion
Release date: February 2, 2023
```

### Additional improvements

- Support for Kubernetes [1.25](https://kubernetes.io/blog/2022/08/23/kubernetes-v1-25-release/) and [1.26](https://kubernetes.io/blog/2022/12/09/kubernetes-v1-26-release/).
- You can now configure custom annotations for Houston ingress by setting `astronomer.houston.ingress.annotation` in your `config.yaml` file. 
- The System Admin **Deployments** list in the Astronomer UI is now paginated. 
- You can now use the following values in your `config.yaml` file to configure resource allocation for the git-sync relay service:
  
    - `astronomer.gitSyncRelay.gitSyncResources`
    - `astronomer.gitSyncRelay.gitDaemonResources`
    - `astronomer.gitSyncRelay.securityContext`

- You can now set `timeoutSeconds` for `readinessProbe` and `livenessProbe` in the Prometheus Helm chart.
- Fixed an issue where Deployments with many DAGs could not be successfully upgraded due to a short timeout.
- Houston API now logs an installation's deployed image versions whenever a GraphQL mutation is completed.

### Bug fixes 

> - Fixed an issue where orphan Deployments and Workspaces remained when users were deactivated with SCIM. 

Reviewer: This is moved to the next patch release. Please omit

> - When SCIM is enabled, service accounts and users can no longer create users through the Houston API. 

Reviewer: This is moved to the next patch release. Please omit

0.31 is the latest stable version of Astronomer Software, while 0.30 remains the latest long-term support (LTS) release. To upgrade to 0.31, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.31.2

Release date: January 31, 2023

### Additional improvements

> - Added support for [Kubernetes 1.25](https://kubernetes.io/blog/2022/08/23/kubernetes-v1-25-release/).

Reviewer 2: ```suggestion
- Support for [Kubernetes 1.25](https://kubernetes.io/blog/2022/08/23/kubernetes-v1-25-release/).
```

Reviewer 3: FWIW we also added support for k8s 1.26

> - You can now configure custom annotations for Houston ingress by setting `astronomer.houston.ingress.annotation` in your `config.yaml` file. 

Reviewer 2: ```suggestion
- You can now use `astronomer.houston.ingress.annotation` in your `config.yaml` file to configure custom annotations for Houston ingress. 
```

> - The System Admin **Deployments** list in the Astronomer UI is now paginated. 

Reviewer 2: ```suggestion
- The System Admin **Deployments** list in the Cloud UI is now paginated. 
```

> - PgBouncer now utilizes graceful shutdowns.

Reviewer 2: ```suggestion
- PgBouncer now gracefully shuts down.
```

Reviewer 3: We've removed this feature from this release.

> - Errors related to the git-sync Kubernetes pod now appear in the Astro UI (potentially not in release)

Author: Saw this one in the issues list but with no attached PR. Did we make this change?

Reviewer 3: This was moved out of this release.

> - Errors related to the git-sync Kubernetes pod now appear in the Astro UI (potentially not in release)

Reviewer 2: ```suggestion
- Errors related to the git-sync Kubernetes pod now appear in the Cloud UI (potentially not in release)
```

> - You can now tune the git-sync relay service's resource usage by setting the following values in your `config.yaml` file:

Reviewer 2: ```suggestion
- You can now use the following values in your `config.yaml` file to configure resource allocation for the git-sync relay service:
```

    - `astronomer.gitSyncRelay.gitSyncResources`
    - `astronomer.gitSyncRelay.gitDaemonResources`
    - `astronomer.gitSyncRelay.securityContext`

> - You can now set `timeoutSeconds` for both `readinessProbe` and `livenessProbe` in the Prometheus Helm chart.

Reviewer 2: ```suggestion
- You can now set `timeoutSeconds` for `readinessProbe` and `livenessProbe` in the Prometheus Helm chart.
```

> - Fixed an issue where Deployments with many DAGs could not be successfully upgraded due to a short timeout.

Reviewer 2: ```suggestion
-  Deployments with many DAGs are now upgraded after a short timeout.
```

Reviewer 3: The suggested wording is incorrect. The problem was that DAGs were not upgraded because the timeout was too short. The timeout has been lengthened.

> - Houston API now logs an installation's deployed image versions whenever a GraphQL mutation is completed.

Reviewer 2: ```suggestion
- The Houston API now logs a deployed image version for installations whenever a GraphQL mutation is completed.
```

Reviewer 3: This suggested wording is incorrect. The API doesn't log the image version for installations. Whenever a GraphQL mutation is completed (which happens frequently during normal operation) houston logs the running docker image versions.

### Bug fixes 

> - Fixed an issue where a user deactivated through SCIM would leave behind orphan Deployments and Workspaces. 

Reviewer 2: ```suggestion
- Fixed an issue where orphan Deployments and Workspaces remained when users were deactivated with SCIM. 
```

> - Service accounts and users can no longer create users through the Houston API when SCIm is enabled. 

Reviewer 3: I think it is. When SCIM is enabled, houston-api will no longer allow users or service accounts to create users.

> - Database migrations for a Deployment now use the same amount of resources as configured for the Deployment's scheduler. This should limit OOM errors when migrating large DAGs.

Reviewer 2: ```suggestion
- To limit Out of Memory (OOM) errors when migrating large DAGs, Deployment database migrations now use the same resources that are configured for the Deployment's scheduler. .
```
